src/Pos/Genesis.hs

```diff
@@ -136,49 +159,36 @@ genesisUtxo bootStakeholdersMaybe ad
     balances = concatMap (uncurry zip) somethingComplicated
     utxoEntry (addr, coin) =
         ( TxIn (unsafeHash addr) 0
--- Note: it's quite bad because HD wallets become stakeholders and each
--- transaction will give them quite a lot of stake.
--- But CSL-1351 is planned for this sprint, so I suppose it's fine (it's
--- unlikely to cause problems anyway).
--- TODO [CSL-1351] Delete ↓.
         , TxOutAux (TxOut addr coin) (outDistr coin))
     -- Empty distribution for PubKey address means that the owner of
     -- this address will have the stake.
-    outDistr coin = maybe [] (flip genesisSplitBoot coin) bootStakeholdersMaybe
--- TODO [CSL-1351] Uncomment ↓.
---         , TxOutAux (TxOut addr coin) (genesisSplitBoot bootStakeholders coin))
---     bootStakeholdersCalculated = balancesToStakeholders balances
---     bootStakeholders =
---         fromMaybe bootStakeholdersCalculated bootStakeholdersMaybe
-
--- balancesToStakeholders :: [(Address, Coin)] -> HashMap StakeholderId Word16
--- balancesToStakeholders balances = foldr step mempty balances
```

Comment:
All stakeholders have equal weight ⇒ `genesisSplitBoot` gives them equal stakes. But we want stake to depend on `StakeDistribution`. Even if you explicitly give them correct stakes but also give them equal weights, it's still bad because further transactions will destroy this distribution.
I think my solution was the most reasonable, it solves this problem and doesn't require a caller to generate genesis stakeholders manually.